native/src/main/java/io/ballerina/stdlib/oracledb/utils/ConverterUtils.java

```diff
@@ -78,16 +88,39 @@ public static String convertIntervalYearToMonth(Object value)
     public static String convertIntervalDayToSecond(Object value)
             throws ApplicationError {
         Map<String, Object> fields = getRecordData(value, Constants.Types.OracleDbTypes.INTERVAL_DAY_TO_SECOND);
-        Object dayObject = fields.get(Constants.Types.IntervalDayToSecond.DAYS);
-        Object hourObject = fields.get(Constants.Types.IntervalDayToSecond.HOURS);
-        Object minuteObject = fields.get(Constants.Types.IntervalDayToSecond.MINUTES);
-        Object secondObject = fields.get(Constants.Types.IntervalDayToSecond.SECONDS);
-        String day = getIntervalString(dayObject, Constants.Types.OracleDbTypes.INTERVAL_DAY_TO_SECOND);
-        String hour = getIntervalString(hourObject, Constants.Types.OracleDbTypes.INTERVAL_DAY_TO_SECOND);
-        String minute = getIntervalString(minuteObject, Constants.Types.OracleDbTypes.INTERVAL_DAY_TO_SECOND);
-        String second = getIntervalString(secondObject, Constants.Types.OracleDbTypes.INTERVAL_DAY_TO_SECOND);
-
-        return day + " " + hour + ":" + minute + ":" + second;
+        long days = 0L;
+        long hours = 0L;
+        long minutes = 0L;
+        double seconds = 0.0d;
+        Object dayObj = fields.get(Constants.Types.IntervalDayToSecond.DAYS);
+        Object hourObj = fields.get(Constants.Types.IntervalDayToSecond.HOURS);
+        Object minuteObj = fields.get(Constants.Types.IntervalDayToSecond.MINUTES);
+        Object secondObj = fields.get(Constants.Types.IntervalDayToSecond.SECONDS);
+        long sign = (Long) fields.get(Constants.Types.IntervalDayToSecond.SIGN);
+        if (dayObj != null) {
+            days = (Long) dayObj;
+        }
+        if (hourObj != null) {
+            hours = (Long) hourObj;
+        }
+        if (minuteObj != null) {
+            minutes = (Long) minuteObj;
+        }
+        if (secondObj != null) {
+            seconds = ((BDecimal) secondObj).floatValue();
+        }
```

Comment:
```suggestion
        long days = fields.get(Constants.Types.IntervalDayToSecond.DAYS == null) ? 0L : (Long) fields.get(Constants.Types.IntervalDayToSecond.DAYS);
        long hours = fields.get(Constants.Types.IntervalDayToSecond.HOURS == null) ? 0L : (Long) fields.get(Constants.Types.IntervalDayToSecond.HOURS);
        long minutes = fields.get(Constants.Types.IntervalDayToSecond.MINUTES == null) ? 0L : (Long) fields.get(Constants.Types.IntervalDayToSecond.MINUTES);
        long seconds = fields.get(Constants.Types.IntervalDayToSecond.SECONDS == null) ? 0.0d : ((BDecimal) fields.get(Constants.Types.IntervalDayToSecond.SECONDS)).floatValue();
        long sign = (Long) fields.get(Constants.Types.IntervalDayToSecond.SIGN);
```

Reply:
changed in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/24/commits/30cf658c64e51247ad8a1af64f541eee89fbb0ca